src/Author.php

```diff
@@ -33,13 +28,26 @@ class Author extends AbstractEndpoint {
 	 * @return array The array with the data of the endpoint
 	 */
 	public function endpoint_callback( \WP_REST_Request $request ) {
-		$user_id = $request->get_param( 'id' );
-		$response = [];
-		$user = get_user_by( 'id', $user_id );
+		$params = $request->get_params();
+
+		$id = ( false === $params['id'] ) ? false : absint( $params['id'] );
+
+		$slug = ( false === $params['slug'] ) ? false : trim( $params['slug'], '/' );
+
+		if ( false === $id && false === $slug ) {
+			return new \WP_Error( self::INVALID_PARAMS, 'The request must have either an id or a slug', [ 'status' => 400 ] );
```

Comment:
I think we had the same discussion for post... IMO it's better to reject the request as invalid if the id<=0, rather than let it pass and catch later or ignore (convert to a positive int).

In any case it's what we ended up doing in post and we should be consistent. So I vote we merge this.